src/session_impl.cpp

```diff
@@ -1476,6 +1470,8 @@ namespace aux {
 
 		if (reopen_listen_port)
 		{
+			if (init)
```

Comment:
this condition could be hoisted out of the reopen_listen_port if-block, right? (you save one nesting level and get slightly simpler logic)